pkg/kubelet/cm/memorymanager/memory_manager.go

```diff
@@ -375,7 +375,7 @@ func validateReservedMemory(machineInfo *cadvisorapi.MachineInfo, nodeAllocatabl
 		}
 
 		if !(*nodeAllocatableMemory).Equal(*reservedMemory) {
-			return fmt.Errorf("the total amount of memory of type \"%s\" is not equal to the value determined by Node Allocatable feature", resourceType)
+			return fmt.Errorf("the total amount %q of type %q is not equal to the value %q determined by Node Allocatable feature", reservedMemory.String(), resourceType, nodeAllocatableMemory.String())
```

Comment:
I don't understand what this check is doing?  Can you explain why nodeAllocatable must equal reserved?

Reply:
The`nodeAllocatable` feature can give you a way to configure some amount of memory to be reserved for the system, kube, or eviction procedure, but it does not provide you a way to granulate the reserved memory among NUMA nodes. To inform the memory manager how much memory should be reserved from each NUMA node we introduced the `--reserved-memory` flag, but this flag does not makes any sense out of the memory manager and should be aligned together with the node allocatable flags because the scheduler makes in account theses flags during allocatable memory under the node.

Reply:
Oh, so --reserved-memory is subdividing another flag.  Hrm.  I kind of expected that to be done in kubelet configuration validation (both flags are available there) not in the memory manager.

Reply:
I has a similar comment early on in the review process: https://github.com/kubernetes/kubernetes/pull/95479#discussion_r522102019

Reply:
The main problem that all relevant `nodeAllocatable` parameters parsed after the `ValidateKubeletConfiguration` called, you can see https://github.com/kubernetes/kubernetes/blob/0e2248f753d0b99cb3ac12bfa48c692c0ebbe70e/cmd/kubelet/app/server.go#L690 and some parameters also depends on the value under kubelet flags(see - https://github.com/kubernetes/kubernetes/blob/0e2248f753d0b99cb3ac12bfa48c692c0ebbe70e/cmd/kubelet/app/server.go#L700)
Probably I can refactor it but I think it should be done under the separate PR.
For the current PR, I can:
- move the validation logic https://github.com/kubernetes/kubernetes/pull/95479/files#diff-501be18894cc46e590062eb7d2a32948635de2de8bcc8baf4a44cf3916dea084R377 under the line https://github.com/kubernetes/kubernetes/blob/0e2248f753d0b99cb3ac12bfa48c692c0ebbe70e/cmd/kubelet/app/server.go#L711
- leave it as it now, and move it, once the refactor PR will be merged

@klueska @smarterclayton WDT?

Reply:
An additional option, I can move this validation to https://github.com/kubernetes/kubernetes/blob/0e2248f753d0b99cb3ac12bfa48c692c0ebbe70e/cmd/kubelet/app/options/options.go#L290

Reply:
After additional digging, the eviction threshold hardly coupled with the capacity of the machine(the kubelet get the capacity from the cadvisor and the cadvisor initialized after the kubeletConfig validation and also after the server validation)

Reply:
Validation is always hermetic (depends on nothing except feature gates and the fields in a given config), but any validation that CAN be done in validation should be.  For instance, if a feature gate makes a calculation more complex, the parts of the calculation that can be done with the gate off could still be in validation.  (so if the config had partial data, we could validate with that partially, then rely on a deeper validation).

That said, this is fine for now.